### PR TITLE
A11Y: Improve select-kit accessibility (take 2)

### PR DESCRIPTION
**Pull request by pmusaraj**
Same as #214500 but fixes an issue with mobile navigation in some cases.

### CODE REVIEW

File: app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js [hunk 1]
```diff
@@ -10,55 +11,53 @@ export default Component.extend({
...
+        this.isDestroying ||
+        this.isDestroyed ||
+        this.selectKit.mainElement()?.contains(document.activeElement)
+      ) {
+        return;
+      }
+
       this.selectKit.close(event);
-    }
+    }, 50);
```
Comment (davidtaylorhq) [inline app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js:61]: This seems like it has the potential to be flaky... what's the significance of 50ms here? (I suspect this delay is the reason that `selectKit.mainElement()` returned null and caused errors)

Comment (pmusaraj) [reply app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js:61]: This is not ideal, agreed. The reason for this delay is that when during `keyDown` the activeElement hasn't changed yet, so we need to wait a bit and then check. I'm looking at alternatives... 

That said, this is not the source of the errors. There was no keydown event happening at all during these errors, I can repro the issue locally even when I comment out the full `later()` block. The issue is that the event is assigned to the parent of `select-kit-body` and calling it via `this.selectKit.mainElement()` seems to be slow, the parent's already removed by then.

Comment (davidtaylorhq) [reply app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js:61]: > during keyDown the activeElement hasn't changed yet

I see :+1:. I wonder if we could use `next()` instead? If it works, that feels a little cleaner than an arbitrary delay.

Comment (pmusaraj) [reply app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js:61]: Yes, just switched to `next()`, it works well!

Comment (pmusaraj) [reply app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-body.js]: I'd like to try merging this again on my tomorrow morning @davidtaylorhq, if there are no further objections. Thanks!

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**